packages/lib/src/vis/shared/PanZoomMesh.tsx

```diff
@@ -3,27 +3,38 @@ import type { ThreeEvent } from '@react-three/fiber/dist/declarations/src/core/e
 import { clamp } from 'lodash';
 import { useRef, useCallback, useEffect } from 'react';
 import type { Vector3 } from 'three';
+import { Vector2 } from 'three';
 
 import { useWheelCapture } from '../hooks';
 import { useAxisSystemContext } from './AxisSystemContext';
 
 const ZOOM_FACTOR = 0.95;
 
 function PanZoomMesh() {
-  const { visSize } = useAxisSystemContext();
+  const { abscissaScale, ordinateScale, visSize } = useAxisSystemContext();
+  const { width: visWidth, height: visHeight } = visSize;
 
   const camera = useThree((state) => state.camera);
   const { width, height } = useThree((state) => state.size);
   const invalidate = useThree((state) => state.invalidate);
 
   const startOffsetPosition = useRef<Vector3>(); // `useRef` to avoid re-renders
+  const viewportCenter = useRef<Vector2>();
 
   const moveCameraTo = useCallback(
-    (x: number, y: number) => {
-      const { position, zoom } = camera;
+    (x: number, y: number, saveCoords = false) => {
+      if (saveCoords) {
```

Comment:
Is `saveCoords` really necessary ? I mean, it does not hurt to save the viewport center once again when restoring it ?

We could also move the `saveCoords` part out of `moveCameraTo` to be put in a dedicated callback ?

Reply:
> Is saveCoords really necessary ? I mean, it does not hurt to save the viewport center once again when restoring it ?

I guess not, good point. It's not triggering any re-renders since it's in a ref, so the impact is minimal.